src/eth/DSProxyService.js

```diff
@@ -67,14 +76,20 @@ export default class DSProxyService extends PrivateService {
       { contract: 'PROXY_REGISTRY', method: 'build' }
     ).mine();
     this._currentProxy = await this.getProxyAddress();
+    this.get('web3')
+      .get('event')
+      .emit('dsproxy/BUILD', {
+        address: this._currentProxy
+      });
     return txo;
```

Comment:
not sure if there's any use in returning the `txo` at this point since we've already awaited its mining? maybe you should just return `this._currentProxy` and then `ensureProxy` can be a little simpler and just return `this.build()`